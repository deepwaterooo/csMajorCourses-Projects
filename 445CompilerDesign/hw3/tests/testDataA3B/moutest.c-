// from Xin Mou
// F13
main();

int emu(int x, y, z)
{                    // Warning: There is no return
}

test(){
    int aa[6], bb[6], cc[6];
    bool dd[6];

    int x;
    bool b;
    x = aa[
        bb[
            cc[
                dd[
                    aa[
                        dd]]]]];
    x = aa[
        bb[
            cc[
                dd[
                    aa[
                        dd[
                            1]]]]]];
    x = aa[
        bb[
            cc[
                dd[
                    aa[
                        *dd]]]]];
    x = aa[
        bb[
            cc[
                dd[
                    aa[
                        *x]]]]];


    emu(false,
        true and emu(x+b,
                     emu(x+b,
                         emu(x+b,
                             parrot,
                             b and x),
                         b and x),
                     b and x),
        true,
        x*b);
}
