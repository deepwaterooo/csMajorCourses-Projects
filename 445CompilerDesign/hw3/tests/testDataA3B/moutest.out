//undef.out

WARNING(3): Expecting to return type int but function 'intvalue' has no return statement.
ERROR(10): Symbol 'undef' is not defined.
ERROR(10): Initializer for variable 'x' is not a constant expression.
ERROR(10): Symbol 'x' is already defined at line 9.
ERROR(14): Symbol 'undef' is not defined.
ERROR(15): '+' requires operands of type int but rhs is of type void.
ERROR(16): Symbol 'undef' is not defined.
ERROR(17): '+' requires operands of type int but lhs is of type void.
ERROR(18): Symbol 'undef' is not defined.
ERROR(18): Symbol 'undef' is not defined.
ERROR(19): '+' requires operands of type int but lhs is of type void.
ERROR(19): '+' requires operands of type int but rhs is of type void.
ERROR(22): Symbol 'undef' is not defined.
ERROR(23): '+' requires operands of type int but rhs is of type void.
ERROR(24): Symbol 'undef' is not defined.
ERROR(25): '+' requires operands of type int but lhs is of type void.
ERROR(26): Symbol 'undef' is not defined.
ERROR(26): Symbol 'undef' is not defined.
ERROR(27): '+' requires operands of type int but lhs is of type void.
ERROR(27): '+' requires operands of type int but rhs is of type void.
ERROR(29): Symbol 'undef' is not defined.
ERROR(30): Symbol 'undef' is not defined.
ERROR(31): Symbol 'undef' is not defined.
ERROR(32): Symbol 'undef' is not defined.
ERROR(34): Symbol 'undef' is not defined.
ERROR(34): '+' requires operands of type int but rhs is of type bool.
ERROR(34): '==' requires operands of the same type but lhs is type int and rhs is type char.
ERROR(34): The operation '*' only works with arrays.
ERROR(35): 'x' is a simple variable and cannot be called.
ERROR(35): '+' requires operands of type int but rhs is of type bool.
ERROR(35): '==' requires operands of the same type but lhs is type int and rhs is type char.
ERROR(35): The operation '*' only works with arrays.
ERROR(37): Cannot use function 'novalue' as a simple variable.
ERROR(37): '=' requires operands of the same type but lhs is type void and rhs is type int.
ERROR(38): Cannot use function 'intvalue' as a simple variable.
ERROR(40): Expecting type int in parameter 1 of call to 'take1' defined on line 1 but got type void.
ERROR(41): Cannot use function 'novalue' as a simple variable.
ERROR(41): Expecting type int in parameter 1 of call to 'take1' defined on line 1 but got type void.
ERROR(42): Symbol 'undef' is not defined.
ERROR(43): Symbol 'undef' is not defined.
ERROR(45): Cannot use function 'intvalue' as a simple variable.
Number of warnings: 1
Number of errors: 40





















//moutest.out

WARNING(5): Expecting to return type int but function 'emu' has no return statement.
ERROR(19): Array 'aa' should be indexed by type int but got type bool.
ERROR(20): Array index is the unindexed array 'dd'.
ERROR(17): Array 'cc' should be indexed by type int but got type bool.
ERROR(25): Array 'aa' should be indexed by type int but got type bool.
ERROR(23): Array 'cc' should be indexed by type int but got type bool.
ERROR(30): Array 'cc' should be indexed by type int but got type bool.
ERROR(39): The operation '*' only works with arrays.
ERROR(36): Array 'cc' should be indexed by type int but got type bool.
ERROR(42): Expecting type int in parameter 1 of call to 'emu' defined on line 5 but got type bool.
ERROR(43): '+' requires operands of type int but rhs is of type bool.
ERROR(44): '+' requires operands of type int but rhs is of type bool.
ERROR(45): '+' requires operands of type int but rhs is of type bool.
ERROR(46): Symbol 'parrot' is not defined.
ERROR(47): 'and' requires operands of type bool but rhs is of type int.
ERROR(45): Expecting type int in parameter 3 of call to 'emu' defined on line 5 but got type bool.
ERROR(48): 'and' requires operands of type bool but rhs is of type int.
ERROR(44): Expecting type int in parameter 3 of call to 'emu' defined on line 5 but got type bool.
ERROR(49): 'and' requires operands of type bool but rhs is of type int.
ERROR(43): Expecting type int in parameter 3 of call to 'emu' defined on line 5 but got type bool.
ERROR(43): 'and' requires operands of type bool but rhs is of type int.
ERROR(42): Expecting type int in parameter 2 of call to 'emu' defined on line 5 but got type bool.
ERROR(42): Expecting type int in parameter 3 of call to 'emu' defined on line 5 but got type bool.
ERROR(42): Too many parameters passed for function 'emu' defined on line 5.
ERROR(51): '*' requires operands of type int but rhs is of type bool.
Number of warnings: 1
Number of errors: 24

